internal/v2/fakes2av2/fakes2av2.go

```diff
@@ -0,0 +1,115 @@
+package fakes2av2
+
+import (
+	"log"
+	"fmt"
+	_ "embed"
+	s2av2pb "github.com/google/s2a-go/internal/proto/v2/s2a_go_proto"
+	commonpb "github.com/google/s2a-go/internal/proto/v2/common_go_proto"
+)
+var (
+	//go:embed example_cert/client_cert.pem
+	clientCert []byte
+	//go:embed example_cert/server_cert.pem
+	serverCert []byte
+)
+
+// Server is a Fake S2A Server for testing.
+type Server struct {
+	s2av2pb.UnimplementedS2AServiceServer
+}
+
+// SetUpSession receives SessionReq, performs request, and returns a
+// SessionResp, all on the server stream.
+func (s *Server) SetUpSession(stream s2av2pb.S2AService_SetUpSessionServer) error {
+	for {
+		req, err := stream.Recv()
+		if err != nil {
+			log.Printf("Fake S2A Service: failed to receive SessionReq: %v", err)
+			return err
+		}
+		// Call one of the 4 possible RespOneof's
+		// TODO(rmehta19): Consider validating the body of the request.
+		var resp *s2av2pb.SessionResp
+		var err0 error
```

Comment:
nit: we don't need to declare err0 here, we can just redefine the error each time we get one